app/forms/validators.py

```diff
@@ -49,15 +54,15 @@ def __call__(
         field: Union[DecimalFieldWithSeparator, IntegerFieldWithSeparator],
     ) -> None:
         try:
-            Decimal(
-                field.raw_data[0].replace(
-                    numbers.get_group_symbol(flask_babel.get_locale()), ""
-                )
-            )
+            Decimal(sanitise_number(number=field.raw_data[0]))
         except (ValueError, TypeError, InvalidOperation, AttributeError) as exc:
             raise validators.StopValidation(self.message) from exc
 
-        if "e" in field.raw_data[0].lower():
+        if "e" in field.raw_data[0].lower() or math.isnan(
+            Decimal(
+                sanitise_number(number=field.raw_data[0])
+            )  # number is sanitised to guard against inputs like `,NaN_` etc
```

Comment:
The comment shouldn't be here. Add doc strings to the sanitise number method instead.

Reply:
I was thinking this comment only applies here when the value entered is `",NaN_"` or similar. If we don't  use sanitise method here validation would not stop, but it is such a rare case I preferred to leave comment.

Reply:
Fair point.